src/components/checkout-button.ts

```diff
@@ -0,0 +1,127 @@
+import { LitElement, html, property, customElement, css} from 'lit-element';
+import {buttonStyle} from '../styles/buttons'
+import {typographyStyle} from '../styles/typography'
+import { Product, OrderMeta, Order, CheckoutOptions } from '../types/shop';
+import HubApi from '@nimiq/hub-api';
+import { DummyShopStorage, IpfsShopStorage, ShopStorage } from '../storage';
+
+enum SubmitState {
+  NotSubmitted,
+  Submitting,
+  SubmitSucceed,
+  SubmitFailed
+}
+
+@customElement('nimiq-shop-checkout-button')
+export class CheckoutButton extends LitElement {
+
+    static get styles() {
+      return [
+        buttonStyle,
+        typographyStyle
+        , css`:host ul {
+          list-style: none;
+        }`
+      ]
+    }
+
+  // will be set by 'nimiq-shop'
+  @property({type: String, attribute: 'shop-address'}) 
+  shopAddress = '';
+
+  @property({type: String}) 
+  label = '';
+
+  protected get hasLabel(): Boolean {
+    return this.label != '' && this.label.length > 0
+  }
+
+  @property({type: String}) 
+  name = null;
+
+  @property({type: String, attribute: 'button-class'}) 
+  buttonClass = 'nq-button';
+  
+  @property({type: Number}) 
+  price = 0;
+  
+  @property({type: SubmitState}) 
+  submitState = SubmitState.NotSubmitted;
+
+  // error messages
+  errors: String[] = []
+
+  // Nimiq Hub Api
+  hubApi: HubApi = new HubApi(process.env.HUB_API)
+
+  // Storage
+  // TODO(sectore): Put 'repo' into storage
+  storage: ShopStorage = process.env.production ? 
+    new IpfsShopStorage('') : new DummyShopStorage('')
```

Comment:
would make this a static method on `ShopStorage` (to reuse it)

Reply:
`ShopStorage` is a TS `interface` now: https://github.com/svub/nimiq-popup-shop/blob/84dd10778e8479bfa7cd05ef69fd3a3d1168cacc/src/storage.ts#L4-L8 It can't have a static method anymore. If we will see that we do need this check at other places as well, then we can extract that into a helper function or so.